src/gui_common/tooltip/ToolTipManager.cs

```diff
@@ -296,33 +321,56 @@ private void FetchToolTips()
 
             foreach (ICustomToolTip tooltip in group.GetChildren())
             {
-                tooltip.ToolTipVisible = false;
+                tooltip.ToolTipNode.Visible = false;
                 collectedTooltips.Add(tooltip);
             }
 
             tooltips.Add(group, collectedTooltips);
         }
     }
 
-    private void UpdateToolTipVisibility()
+    private void FinalizeToolTipVisibility(ICustomToolTip tooltip, bool visible)
     {
-        // TODO: Fix the current tooltip changing while still fading out
-        // when quickly mousing over multiple closely positioned elements
-        // (Happens when a single tooltip is registered to multiple Controls)
-
-        // Make sure to hide any other tooltips that are still visible
-        HideAllToolTips();
+        if ((tooltip.ToolTipNode.Visible && visible) || (!tooltip.ToolTipNode.Visible && !visible))
+            return;
 
-        if (Display)
+        switch (tooltip.TransitionType)
         {
-            // Set timer
-            displayTimer = MainToolTip.DisplayDelay;
+            case ToolTipTransitioning.Immediate:
+            {
+                tooltip.ToolTipNode.Visible = visible;
+                break;
+            }
+
+            case ToolTipTransitioning.Fade:
+            {
+                // TODO: Fix fading when tooltip display delay is less than the tooltip fade speed, some kind of
+                // flickering happens
+                if (visible)
+                {
+                    GUICommon.Instance.ModulateFadeIn(tooltip.ToolTipNode, Constants.TOOLTIP_FADE_SPEED);
+                }
+                else
+                {
+                    GUICommon.Instance.ModulateFadeOut(tooltip.ToolTipNode, Constants.TOOLTIP_FADE_SPEED);
+                }
+
+                break;
+            }
```

Comment:
use `case ToolTipTransitioning.Fade when visible:` and `case ToolTipTransitioning.Fade when !visible:` here